### DISCUSSION TITLE
Does WPF support setting text character spacing?

### DISCUSSION THREAD

**Discussion by SetTrend** [dotnet/wpf#10037] [Q&A] [answered]
I'd like to change character spacing of text displayed in a control.

Is this possible using WPF?

Here's an example, taken from the corresponding text parameter in Adobe Illustrator:
![Text character spacing](https://github.com/user-attachments/assets/d3b43df1-474f-4c74-8736-c8e2c87c8c24)

**Comment by miloush** [comment]
Not natively. You would need to use GlyphRun and adjust the spacing yourself.

**Comment by SetTrend** [comment]
Thank you for enlightening me!

This feature seems being a deeply burried font feature in WPF and cumbersome to reach.

Would it make sense to suggest to add letter and word spacing delta properties to the Font properties of `Control` and `TextBlock`? (I.e., `CharacterSpacing` and `WordSpacing`.)

Letter (and word) spacing are a quite common feature in DTP programs, like, for instance, MS Word:
![Font-Screen](https://github.com/user-attachments/assets/ef97a578-e4ed-4864-9b13-22d3b893ae57)

… Adobe products (see above), or CorelDRAW:
![CorelDRAW text settings](https://github.com/user-attachments/assets/6a11dc8c-a856-4896-8ca1-431bbeddfa3d)

**Reply by miloush** [reply, marked as answer]
See #293 (and more recently #9868). It would probably go to `TextElement` or `Typopgrahy`. Not sure about word spacing as for some languages it requires linguistic analysis.

**Reply by SetTrend** [reply]
NB: Cross referenced this discussion in #293

**Reply by lindexi** [reply]
@miloush It should be #9868 not #9686
